components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt

```diff
@@ -61,16 +61,18 @@ abstract class GenericScalarStorageEngine<ScalarType> : StorageEngine {
     protected abstract fun deserializeSingleMetric(value: Any?): ScalarType?
 
     /**
-     * Implementor's provided function to serialized 'user' lifetime
-     * data to [String].
+     * Implementor's provided function to serialize 'user' lifetime data as needed by the data type.
      *
-     * @param value loaded from the storage as [ScalarType]
-     *
-     * @return data as [String] or null if serialization failed
+     * @param userPreferences SharedPreference.Editor for writing preferences as needed by type.
```

Comment:
nit: please wrap `SharedPreference.Editor` in square brackets, so that kdoc correctly links to the right class in the documentation. (same for other objects in function/class docs)